src/context/WalletProvider/types.ts

```diff
@@ -0,0 +1,29 @@
+import type { KkRestAdapter } from '@keepkey/hdwallet-keepkey-rest'
+import type { CoinbaseAdapter } from '@shapeshiftoss/hdwallet-coinbase'
+import type { WebUSBKeepKeyAdapter } from '@shapeshiftoss/hdwallet-keepkey-webusb'
+import type { KeplrAdapter } from '@shapeshiftoss/hdwallet-keplr'
+import type { WebUSBLedgerAdapter } from '@shapeshiftoss/hdwallet-ledger-webusb'
+import type { MetaMaskAdapter } from '@shapeshiftoss/hdwallet-metamask'
+import type { NativeAdapter } from '@shapeshiftoss/hdwallet-native'
+import type { WalletConnectV2Adapter } from '@shapeshiftoss/hdwallet-walletconnectv2'
+import type { XDEFIAdapter } from '@shapeshiftoss/hdwallet-xdefi'
+
+import type { KeyManager } from './KeyManager'
+
+export type AdaptersByKeyManager = {
+  [KeyManager.Mobile]: [NativeAdapter]
+  [KeyManager.Native]: [NativeAdapter]
+  [KeyManager.Demo]: [NativeAdapter]
+  [KeyManager.KeepKey]: [KkRestAdapter | typeof WebUSBKeepKeyAdapter]
+  [KeyManager.Ledger]: [WebUSBLedgerAdapter]
+  [KeyManager.Keplr]: [KeplrAdapter]
+  [KeyManager.WalletConnectV2]: [WalletConnectV2Adapter]
+  [KeyManager.MetaMask]: [MetaMaskAdapter]
+  [KeyManager.Coinbase]: [CoinbaseAdapter]
+  [KeyManager.XDefi]: [XDEFIAdapter]
```

Comment:
is this to say, the values are an array but never more than of length 1?

Reply:
Not necessarily, e.g KK has two adapters - MM is also a union of two adapters (never more than one at the same time however) programmatically on the flag being turned on/off, though we simply it as `MetaMaskAdapter` because of type shenanigans here

Reply:
lets discuss this, it doesnt make sense to have a tupleof length 1, so we should remove the concept of an array and reduce the complexity of the surrounding code